crypto/default/src/main/java/org/keycloak/crypto/hash/Argon2PasswordHashProviderFactory.java

```diff
@@ -23,6 +24,13 @@ public class Argon2PasswordHashProviderFactory implements PasswordHashProviderFa
     public static final String ITERATIONS_KEY = "iterations";
     public static final String PARALLELISM_KEY = "parallelism";
 
+    /**
+     * The Argon2 password hashing is CPU bound, so it doesn't make sense to hash more values concurrently than there are cores on the machine.
+     * When we run more, this only leads to an increased memory usage and to throttling of the process in containerized environments
+     * when a CPU limit is imposed. The throttling would have a negative impact on other concurrent non-hashing activities of Keycloak.
+     */
+    private static final Semaphore CPU_CORE_SEMAPHORE = new Semaphore(Runtime.getRuntime().availableProcessors());
```

Comment:
We should probably make this configurable. That could be useful to limit the concurrent hash operations below the number of processors, if there is a large number of processors available.

Reply:
This is now configurable, and reflected in the docs.